zulipterminal/helper.py

```diff
@@ -618,3 +619,9 @@ def display_error_if_present(response: Dict[str, Any], controller: Any
                              ) -> None:
     if response['result'] == 'error' and hasattr(controller, 'view'):
         controller.view.set_footer_text(response['msg'], 3)
+
+
+def hash_util_decode(string: str) -> str:
+    # Acknowledge custom string replacements in zulip/zulip's
+    # zerver/lib/url_encoding.py before unquote.
+    return unquote(string.replace('.', '%').replace('%2E', '.'))
```

Comment:
I seem to recall a discussion about wanting decoding of server URLs previously on czo - might be worth a check, maybe it's something that could be added to the python API as a utility?

Reply:
(Being discussed in [#integrations > Decode server URLs](https://chat.zulip.org/#narrow/stream/127-integrations/topic/Decode.20server.20URLs).)